apps/MCF/mcf_sparse_matrix.cuh

```diff
@@ -49,4 +170,102 @@ __global__ static void mcf_A_B_setup(
     query.dispatch<Op::VV>(block, shrd_alloc, init_lambda);
 }
 
-void mcf_rxmesh_solver() {}
\ No newline at end of file
+template <typename T>
+void mcf_rxmesh_cusolver_chol(rxmesh::RXMeshStatic&              rxmesh,
+                              const std::vector<std::vector<T>>& ground_truth)
+{
+    using namespace rxmesh;
+    constexpr uint32_t blockThreads = 256;
+
+    uint32_t num_vertices = rxmesh.get_num_vertices();
+    auto     coords       = rxmesh.get_input_vertex_coordinates();
+
+    SparseMatrix<float> A_mat(rxmesh);
+    DenseMatrix<float>  X_mat(num_vertices, 3);
+    DenseMatrix<float>  B_mat(num_vertices, 3);
+
+    printf("use_uniform_laplace: %d, time_step: %f\n",
+           Arg.use_uniform_laplace,
+           Arg.time_step);
+
+    // B set up
+    LaunchBox<blockThreads> launch_box_B;
+    rxmesh.prepare_launch_box({Op::VV},
+                              launch_box_B,
+                              (void*)mcf_B_setup<float, blockThreads>,
+                              !Arg.use_uniform_laplace);
+
+    mcf_B_setup<float, blockThreads><<<launch_box_B.blocks,
+                                       launch_box_B.num_threads,
+                                       launch_box_B.smem_bytes_dyn>>>(
+        rxmesh.get_context(), *coords, B_mat, Arg.use_uniform_laplace);
+
+    CUDA_ERROR(cudaDeviceSynchronize());
+
+    // A and X set up
+    LaunchBox<blockThreads> launch_box_A_X;
+    rxmesh.prepare_launch_box({Op::VV},
+                              launch_box_A_X,
+                              (void*)mcf_A_X_setup<float, blockThreads>,
+                              !Arg.use_uniform_laplace);
+
+    mcf_A_X_setup<float, blockThreads>
+        <<<launch_box_A_X.blocks,
+           launch_box_A_X.num_threads,
+           launch_box_A_X.smem_bytes_dyn>>>(rxmesh.get_context(),
+                                            *coords,
+                                            A_mat,
+                                            X_mat,
+                                            Arg.use_uniform_laplace,
+                                            Arg.time_step);
+
+    // Solving the linear system using chol factorization and no reordering
+    A_mat.spmat_linear_solve(B_mat, X_mat, Solver::CHOL, Reorder::NONE);
+
+    auto smooth_X =
+        rxmesh.add_vertex_attribute<T>("smooth_X", 3, rxmesh::LOCATION_ALL);
+    auto truth_X =
+        rxmesh.add_vertex_attribute<T>("truth_X", 3, rxmesh::LOCATION_ALL);
```

Comment:
I don't think you need this variable `truth_X`. it is only allocated here, the moved from device to host, and then the ground_truth is written into it.

Reply:
Yes. I don't need it. I used it for temporary debugging purpose.